invokeai/backend/install/model_install_backend.py

```diff
@@ -416,13 +418,17 @@ def _download_hf_pipeline(self, repo_id: str, staging: Path) -> Path:
         does a save_pretrained() to the indicated staging area.
         """
         _, name = repo_id.split("/")
-        revisions = ["fp16", "main"] if self.config.precision == "float16" else ["main"]
+        precision = torch_dtype(choose_torch_device())
+        revisions = ["fp16", "main"] if precision == torch.float16 else ["main"]
         model = None
         for revision in revisions:
             try:
-                model = DiffusionPipeline.from_pretrained(repo_id, revision=revision, safety_checker=None)
-            except:  # most errors are due to fp16 not being present. Fix this to catch other errors
-                pass
+                model = DiffusionPipeline.from_pretrained(
+                    repo_id, revision=revision, safety_checker=None, torch_dtype=precision
+                )
```

Comment:
I think using `revision` to select model weight is outdated; see [Patrick's comments](https://discord.com/channels/1020123559063990373/1068277893366546463/1133899572268249108).

https://huggingface.co/docs/diffusers/using-diffusers/loading#checkpoint-variants

https://github.com/huggingface/diffusers/blob/6c5bd2a38d2d07603f17d60e87c009710d251990/src/diffusers/pipelines/pipeline_utils.py#L188

The old `revision` method might work on the SD 1.5 repo since it's been around for longer, but the SDXL repo has no such revision.

Reply:
Fortunately this doesn't hurt. If there is no fp16 revision then the code will try main. The real problem was that `from_pretrained()` was not setting `torch_dtype` appropriately, so that even 16-bit models were being upcast to 32 bit during the subsequent `save_pretrained()`. There are still enough `fp16` revision models around that this should remain in the code.

For next revision of model manager, we'll get rid of the whole `from_pretrained()` method and replace with direct interrogation and downloading of models from HF. This will also remove dependency on the HF hub.

Reply:
If the HF interface is not working correctly, would it maybe make more sense to fix it and push the fix upstream to HF?
I'm asking this without any context, so I could be way off. It just seems better to have HF maintain their own API and then use it. Maybe there are other considerations that I am not aware of

Reply:
The HF interface is working fine; no upstream fixes needed. However, we want to create a threaded model download manager that sends status updates to the front end, and to get this to work we will use the HF API to retrieve the URLs of the model folders and submodels and download them ourselves directly without using the cache.